persist-cli/src/main/java/io/ballerina/persist/cmd/Push.java

```diff
@@ -128,14 +131,14 @@ public void execute() {
             balProject.currentPackage().getCompilation();
             persistConfigurations = SyntaxTreeGenerator.readPersistToml(Paths.get(this.sourcePath, PERSIST,
                     PERSIST_TOML_FILE));
-            int templatedEntryCount = 0;
+            ArrayList<String> templatedEntry = new ArrayList<>();
             for (String key : persistConfigurations.keySet()) {
                 if (Pattern.matches(PLACEHOLDER_PATTERN, persistConfigurations.get(key))) {
-                    templatedEntryCount += 1;
+                    templatedEntry.add(key);
                 }
             }
-            if (templatedEntryCount > 0) {
-                populatePlaceHolder(persistConfigurations);
+            if (!templatedEntry.isEmpty()) {
+                populatePlaceHolder(persistConfigurations, templatedEntry);
```

Comment:
```suggestion
            if (!templatedEntry.isEmpty()) {
                Map<String,String> resolvedEntries= resolveTemplatedConfigurations(templatedEntry);
                persistConfigurations.add(resolvedEntries);
```
Change templatedEntry(templatedEntries) to a map. Easier to handle.

Reply:
updated in [df30e70](https://github.com/ballerina-platform/persist-tools/pull/65/commits/df30e70bcabe76fa78dc959280352bb2f08d8678)